### PR TITLE
Add GUI for task management

### LINKED ISSUE #89: Feature: GUI for task management

### PR DESCRIPTION
**Pull request by Ugholaf**
# GUI for task management
- The basic design for the task management GUI.
- `MainDisplay` will display both the `PersonListPanel` and `TodoListPanel` side by side.
- `TodoCard` will display the description, priority and DateTime of the todo.
- `TodoListPanel` has a header text which depends on the command input, if the user searches the todo by tag, the header will display the given tag name, if the user searches by date, it will be the inputted date.

# Note
This might not be the final design of the GUI, will make changes again after the `Todo` class is completed and tested with some sample data.

image

### CODE REVIEW

File: src/main/resources/view/TodoListCard.fxml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+<?import javafx.scene.control.Label?>
+<?import javafx.scene.layout.ColumnConstraints?>
+<?import javafx.scene.layout.FlowPane?>
+<?import javafx.scene.layout.GridPane?>
+<?import javafx.scene.layout.HBox?>
+<?import javafx.scene.layout.Region?>
+<?import javafx.scene.layout.RowConstraints?>
+<?import javafx.scene.layout.VBox?>
+
+<HBox id="todoCardPane" fx:id="todoCardPane" xmlns="http://javafx.com/javafx/18" xmlns:fx="http://javafx.com/fxml/1">
```
Comment (PangKuangWei) [inline src/main/resources/view/TodoListCard.fxml]:
```suggestion
<HBox id="todoCardPane" fx:id="todoCardPane" xmlns="http://javafx.com/javafx/8" xmlns:fx="http://javafx.com/fxml/1">
```

File: src/main/resources/view/TodoListPanel.fxml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<?import javafx.scene.control.Label?>
+<?import javafx.scene.control.ListView?>
+<?import javafx.scene.layout.VBox?>
+
+<VBox xmlns="http://javafx.com/javafx/18" xmlns:fx="http://javafx.com/fxml/1" alignment="CENTER">
```
Comment (PangKuangWei) [inline src/main/resources/view/TodoListPanel.fxml]:
```suggestion
<VBox xmlns="http://javafx.com/javafx/8" xmlns:fx="http://javafx.com/fxml/1" alignment="CENTER">
```

File: src/main/java/soconnect/ui/TodoCard.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    /**
+     * Creates a {@code TodoCard} with the given {@code Todo} and index to display.
+     */
+    public TodoCard(Todo todo, int displayedIndex) {
+        super(FXML);
+        this.todo = todo;
+        id.setText(displayedIndex + ". ");
+        description.setText(todo.getDescription());
+        datetime.setText(todo.getDatetime());
+        priority.getChildren().add(new Label(todo.getPriority()));
```
Comment (PangKuangWei) [inline src/main/java/soconnect/ui/TodoCard.java:38]: If you want the font style and font size to match with the one set in css, can manually use `setStyle()` on the label object.

**Review by PangKuangWei — CHANGES_REQUESTED** [review]
Do change the javafx version to 8 in the fxml files to avoid warnings when launching the app. Otherwise, LGTM.

File: src/main/java/soconnect/ui/TodoCard.java (hunk 3 above)
Comment (Ugholaf) [reply src/main/java/soconnect/ui/TodoCard.java:38]: Yup thanks for the suggestion, I will update the style once we tested with some sample data.

**Pull request merged**